src/Raven.Server/Documents/Indexes/MapReduce/ReduceKeyProcessor.cs

```diff
@@ -217,15 +199,37 @@ public void Process(object value)
                 return;
             }
 
-            var dynamicArray = value as DynamicArray;
+            var dynamicJson = value as DynamicBlittableJson;
 
-            if (dynamicArray != null)
+            if (dynamicJson != null)
             {
+                var obj = dynamicJson.BlittableJson;
+
+                _mode = Mode.MultipleValues;
+
                 if (_buffer == null)
                     _buffer = _buffersPool.Allocate(16);
+                
+                for (int i = 0; i < obj.Count; i++)
+                {
+                    // this call ensures properties to be returned in the same order, regardless their storing order
```

Comment:
Getting them by index id would give them to you in the lexical order, you are correct.